JaxbProcessor should not register package-info to be bound to JaxbContext

### Describe the bug

Commit [Allow to provide custom configuration for JAXB context](https://github.com/quarkusio/quarkus/commit/e812d24038587981ea1a1c2d50ceb8434c6159b0 ) , introduced  changes to the JaxbProcessor. One of the changes results in the inclusion of **package-info**s to  be bound to the JaxbContext
``` java
  for (AnnotationInstance xmlSchemaInstance : index.getAnnotations(XML_SCHEMA)) {
            if (xmlSchemaInstance.target().kind() == Kind.CLASS) {
                reflectiveClass.produce(
                        new ReflectiveClassBuildItem(false, false, xmlSchemaInstance.target().asClass().name().toString()));
                String className = xmlSchemaInstance.target().asClass().name().toString();

                reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, className));

                classesToBeBound.add(className);
            }
        }
```



### Expected behavior

_No response_

### Actual behavior

Including camel's extensions via **quarkus-camel-bom** results in exceptions such as the following:

Caused by: com.sun.xml.bind.v2.runtime.IllegalAnnotationsException: 21 counts of IllegalAnnotationExceptions
org.apache.camel.model.errorhandler.package-info is an interface, and JAXB can't handle interfaces.
	this problem is related to the following location:
		at org.apache.camel.model.errorhandler.package-info
org.apache.camel.model.config.package-info is an interface, and JAXB can't handle interfaces.
	this problem is related to the following location:
		at org.apache.camel.model.config.package-info
Two classes have the same XML type name "{http://camel.apache.org/schema/spring}packageInfo". Use @XmlType.name and @XmlType.namespace to assign different names to them.
	this problem is related to the following location:
		at org.apache.camel.model.errorhandler.package-info
	this problem is related to the following location:
		at org.apache.camel.model.config.package-info

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

openjdk version "11.0.14" 2022-01-18 OpenJDK Runtime Environment GraalVM CE 22.0.0.2 (build 11.0.14+9-jvmci-22.0-b05) OpenJDK 64-Bit Server VM GraalVM CE 22.0.0.2 (build 11.0.14+9-jvmci-22.0-b05, mixed mode, sharing)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.11.1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.4 (9b656c72d54e5bacbed989b64718c159fe39b537)

### Additional information

_No response_

--- a/extensions/jaxb/deployment/src/main/java/io/quarkus/jaxb/deployment/JaxbProcessor.java
+++ b/extensions/jaxb/deployment/src/main/java/io/quarkus/jaxb/deployment/JaxbProcessor.java
@@ -211,8 +211,6 @@ class JaxbProcessor {
                 String className = xmlSchemaInstance.target().asClass().name().toString();
 
                 reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, className));
-
-                classesToBeBound.add(className);
             }
         }
 

--- a/extensions/jaxb/deployment/src/test/java/io/quarkus/jaxb/deployment/info/package-info.java
+++ b/extensions/jaxb/deployment/src/test/java/io/quarkus/jaxb/deployment/info/package-info.java
@@ -1,0 +1,10 @@
+/**
+ * The inclusion of this file is to verify that the produced JAXBContext works fine with it, otherwise the build fails.
+ */
+@XmlSchema(namespace = "http://abc.com", xmlns = {
+        @XmlNs(prefix = "abc", namespaceURI = "http://abc.com")
+})
+package io.quarkus.jaxb.deployment.info;
+
+import javax.xml.bind.annotation.XmlNs;
+import javax.xml.bind.annotation.XmlSchema;

